src/workerd/util/mimetype.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <kj/common.h>
+#include <kj/exception.h>
+#include <kj/map.h>
+#include <kj/string.h>
+
+namespace workerd {
+
+class MimeType final {
+public:
+  using MimeParams = kj::HashMap<kj::String, kj::String>;
+
+  enum ParseOptions {
+    DEFAULT,
+    IGNORE_PARAMS,
+  };
+
+  static kj::Maybe<MimeType> tryParse(kj::StringPtr input,
+                                      ParseOptions options = ParseOptions::DEFAULT);
+  // Returning nullptr implies that the input is not a valid mime type construction.
+  // If the ParseOptions::IGNORE_PARAMS option is set then the mime type parameters
+  // will be ignored and will not be included in the parsed result.
+
+  static MimeType parse(kj::StringPtr input,
+                        ParseOptions options = ParseOptions::DEFAULT);
+  // Asserts if the input could not be parsed as a valid MimeType. tryParse should
+  // be preferred for most cases.
+
+  explicit MimeType(kj::StringPtr type,
+                    kj::StringPtr subtype,
+                    kj::Maybe<MimeParams> params = nullptr);
+
+  MimeType(MimeType&&) = default;
+  MimeType& operator=(MimeType&&) = default;
+  KJ_DISALLOW_COPY(MimeType);
+
+  kj::StringPtr type() const;
+  kj::StringPtr subtype() const;
+
+  const MimeParams& params() const;
+
+  bool set_type(kj::StringPtr type);
```

Comment:
Is there significance in `set_type` and `set_subtype` rather than `setType` and `setSubtype`? Just flagging as it differs without other naming conventions.

Reply:
heh, no, just habbit. Was working on Node.js code also and they use snake case for these and had that stuck in my head.